sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -831,13 +831,21 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       }.toSeq)
     assert(df.schema.map(_.name) === Seq("key", "value", "newCol"))
 
-    val df2 = testData.toDF().withColumn("newCol", col("key") + 1, false)
+    val df2 = testData.toDF().withColumn("newCol", col("key") + 1, 0)
```

Comment:
What about tests with negative positions?

Reply:
Test with negative position was covered for the public method `withColumn` and the private method `withColumns`:
- https://github.com/apache/spark/pull/22332/files#diff-5d2ebf4e9ca5a990136b276859769289R852
- https://github.com/apache/spark/pull/22332/files#diff-5d2ebf4e9ca5a990136b276859769289R907
 
I'm testing 3 cases (in the same time) that add new column at the end:
- negative position
- last position
- position greater than columns size